src/server_manager/install_scripts/install_server.sh

```diff
@@ -81,18 +94,59 @@ function log_for_sentry() {
 # Check to see if docker is installed.
 function verify_docker_installed() {
   if ! command_exists docker; then
-    log_error "Docker CE must be installed, please run \"curl -sS https://get.docker.com/ | sh\" or visit https://docs.docker.com/install/"
-    exit 1
+    log_error "FAILED"
+    echo -n "> Would you like to install Docker? [Y/n] "
+    if ! run_step_with_user_confirmation "Installing Docker" install_docker; then
+      log_error "Docker installation failed, please visit https://docs.docker.com/install for instructions."
+      exit 1
+    fi
+    echo -n "> Verifying Docker installation................ "
+    command_exists docker
   fi
 }
 
 function verify_docker_running() {
-  if ! docker info > /dev/null 2>&1 ; then
-    log_error "It seems like you may not have permission to run Docker.  To solve this, you may need to add your user to the docker group. We recommend running \"sudo usermod -a -G docker $USER && newgrp docker\" and then attempting to install again."
-    exit 1
+  local readonly STDERR_OUTPUT
+  STDERR_OUTPUT=$(sg docker -c "docker info 2>&1 >/dev/null")
+  local readonly RET=$?
+  if [[ $RET -eq 0 ]]; then
+    return 0
+  elif [[ $STDERR_OUTPUT = *"Is the docker daemon running"* ]]; then
+    start_docker
+  fi
+}
+
+function verify_docker_permissions() {
+  local readonly STDERR_OUTPUT
+  STDERR_OUTPUT=$(docker info 2>&1 >/dev/null)
+  local RET=$?
+  if [[ $RET -eq 0 ]]; then
+    return 0
+  elif [[ $STDERR_OUTPUT = *"permission denied"* ]]; then
+    log_error "FAILED"
+    echo -n "> It seems like you may not have permission to run Docker. To solve this, we will attempt to add your user to the docker group. Would you like to proceed? [Y/n] "
```

Comment:
Maybe list the command we are about to run?

Reply:
Done in d281454.